### PR TITLE
Fixed Tileset save causing crash #12708

### LINKED ISSUE #12708: Convert an empty scene to tileset will result to a Godot crash
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**

Windows 10 64-bit, Godot (2.1.4, 3.0-alpha1, and 3.0-alpha2)

**Issue description:**

I tried to convert an empty scene to a tileset by mistake, but I observed that Godot will crash. To avoid this one, we can prevent it by either adding a popup *Alert!* message like `MeshLibrary` does or disabling the menu when we are face in to an empty scene.

**Steps to reproduce:**

1. Create an empty new scene
2. Convert to tileset
3. Save as `<name>.{tres,res}`

Godot will crash when clicking on the save button.

**Link to minimal example project:**

None

### PR DESCRIPTION
**Pull request by CalebMRichardson**
Fix crash when converting empty scene to TileSet

Closes #12708.

### CODE REVIEW

**Comment by mhilbrunner** [comment]
You probably want to clean that up by squishing those commits :)

**Comment by CalebMRichardson** [comment]
Oops. I'll look up how to do that. Thanks!

**Comment by akien-mga** [comment]
Indeed, [see our documentation](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase) about rebasing branches to squash commits. Please also amend the resulting log to be more descriptive, "Fixed issues #" doesn't say anything to someone reading the changelog. A log like this would be better:
```
Fix crash when converting empty scene to TileSet

Closes #12708.
```

**Comment by CalebMRichardson** [comment]
@akien-mga Thanks for the docs. I've updated the log to be more descriptive and I believe I squashed the commits correctly.

**Comment by CalebMRichardson** [comment]
@akien-mga sorry to bother you could you point me in the right direction on why travis-ci keeps failing me?

capture

File: editor/editor_node.cpp [hunk 1]
```diff
@@ -1932,6 +1932,15 @@ void EditorNode::_menu_option_confirm(int p_option, bool p_confirmed) {
 
 		} break;
 		case FILE_EXPORT_TILESET: {
+
```
Comment (akien-mga) [inline editor/editor_node.cpp]: Unnecessary indentation here (empty line), that's why Travis rejects it.

**Comment by akien-mga** [comment]
You should rebase your branch as you have 5 commits for what should be a single one. If you're not familiar with this workflow, I suggest to read the whole page about our [PR workflow](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html).

**Comment by akien-mga** [comment]
I'll cherry-pick the relevant commit manually, it will be simpler :)

**Comment by CalebMRichardson** [comment]
@akien-mga haha thanks! Sorry for the extra work I kinda got myself stuck and have been trying to panic fix it. I'll be more cautious in the future. Thanks again!

**Comment by akien-mga** [comment]
Merged as a607e6125.

**Comment by akien-mga** [comment]
No problem, git is a tough beast and we all had such weird PRs with tons of unrelated changes a few times :D
If you follow the process outlined in the documentation while using the Git command line, you should be safe (GIt GUIs often use different methods than the CLI behind a common name such as "fetch" or "rebase", so we don't recommend using them as they hide the complexity and cause weird issues).

**Pull request closed without merging**